roles/logstash/tasks/logstash-security.yml

```
@@ -427,94 +389,81 @@
-# support
+- name: Set elasticsearch security-api base url for elasticsearch > 7
+  ansible.builtin.set_fact:
+    security_api_base_url: "https://{{ hostvars[groups['elasticsearch'][0]].ansible_default_ipv4.address }}:{{ elasticstack_elasticsearch_http_port }}/_security/"
+  when: elasticstack_release | int > 7
 
-- name: Put logstash_writer role into Elasticsearch < 8
-  ansible.builtin.command: >
-    curl -T /root/logstash_writer_role --header 'Content-Type: application/json'
-    --cacert {{ elasticstack_ca_dir }}/ca.crt
-    -u elastic:{{ logstash_elasticstack_password.stdout }}
-    https://{{ hostvars[groups['elasticsearch'][0]].ansible_default_ipv4.address }}:{{ elasticstack_elasticsearch_http_port }}/_xpack/security/role/logstash_writer
+- name: Set elasticsearch security-api base url for elasticsearch < 8
+  ansible.builtin.set_fact:
+    security_api_base_url: "https://{{ hostvars[groups['elasticsearch'][0]].ansible_default_ipv4.address }}:{{ elasticstack_elasticsearch_http_port }}/_xpack/security/"
+  when: elasticstack_release | int < 8
+
+- name: Check for logstash_writer role
+  ansible.builtin.uri:
+    url: "{{ security_api_base_url }}role/logstash_writer"
+    ca_path: "{{ elasticstack_ca_dir }}/ca.crt"
+    user: elastic
+    password: "{{ logstash_elasticstack_password.stdout }}"
+  register: check_logstash_writer_role_response
   delegate_to: "{{ elasticstack_ca }}"
+  failed_when: false
   changed_when: false
   no_log: "{{ elasticstack_no_log }}"
   run_once: true
-  when:
-    - logstash_writer_role_present.rc > 0 or logstash_reset_writer_role | bool
-    - elasticstack_release | int < 8
 
-- name: Put logstash_writer role into Elasticsearch > 7
-  ansible.builtin.command: >
-    curl -T /root/logstash_writer_role --header 'Content-Type: application/json'
-    --cacert {{ elasticstack_ca_dir }}/ca.crt
-    -u elastic:{{ logstash_elasticstack_password.stdout }}
-    https://{{ hostvars[groups['elasticsearch'][0]].ansible_default_ipv4.address }}:{{ elasticstack_elasticsearch_http_port }}/_security/role/logstash_writer
+- name: Set logstash_writer_role_present
+  ansible.builtin.set_fact:
+    logstash_writer_role_present: true
+  when: check_logstash_writer_role_response.json.logstash_writer is defined
+
+- name: Put logstash_writer role into Elasticsearch if not present
+  ansible.builtin.uri:
+    url: "{{ security_api_base_url }}role/logstash_writer"
+    ca_path: "{{ elasticstack_ca_dir }}/ca.crt"
+    user: elastic
+    password: "{{ logstash_elasticstack_password.stdout }}"
+    method: PUT
+    headers:
+      Content-Type: application/json
+    body: "{{ lookup('template', 'logstash_writer_role.j2') }}"
+    body_format: json
+  register: put_logstash_writer_role_response
+  when: logstash_writer_role_present is not defined
   delegate_to: "{{ elasticstack_ca }}"
-  changed_when: false
+  failed_when: not put_logstash_writer_role_response.json.role.created
```

```suggestion
  failed_when: not put_logstash_writer_role_response.json.role.created | bool
```

I guess it's returning `true`if it works, doesn't it?

---

Done